semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -159,6 +159,29 @@ trait TextDocumentOps { self: SemanticdbOps =>
 
       locally {
         object traverser extends g.Traverser {
+
+          private var currentTress = List[g.Tree]()
+
+          private def materializedTpe(in: g.Tree, currentStack: List[g.Tree] = currentTress) = {
+            def computeTpe(fromTree: g.Tree) = fromTree.tpe match {
+              case null =>
+                None
+              case _ if !config.symbols.saveMaterializedTypes =>
+                None
+              case _ if currentStack.isEmpty =>
+                None
+              case tpe =>
+                tpe.widen.toSemantic._1
+            }
+
+            currentStack.tail match { // Current tree is head...
+              case (tpeApply @ g.TypeApply(_, _)) :: _ =>
+                computeTpe(tpeApply)
```

Comment:
Can you give an example of where this check is used? Looks like a lot of infrastructure for the sake of these two lines of code, so I'd like to better understand the motivation.

Reply:
In case like `foo[Int](1)` scalac use generic type for `foo` and precise type is stored in `foo[Int]`.
Probably we could keep only last tree but full stack is pretty much the same cost to work with.

I am open for any suggestion how to rewrite it.